### ISSUE TITLE
Confusion about config_dict in your code setting.

### ISSUE THREAD

**Issue by jinweiisgreat** [mashijie1028/Happy-CGCD#9]
First of all, thank you very much for providing the code related to the GCD task under continual learning.

To keep things simple, we are using the CIFAR-10 dataset for reproduction. We noticed that according to the sampling strategy in the comments:
```
'cifar10': {'offline_old_cls_num': 7,
             'offine_prop_train_labels': 0.8,   # offline, ratio of labeled data from old classes
             'continual_session_num': 3,   # num of continual learning sessions
             'online_novel_unseen_num': 4000,   # each continual session: num of samples per novel (unseen & first-time) class
             'online_old_seen_num': 250,   # each continual session: num of samples per old (labeled) class
             'online_novel_seen_num': 250,   # each continual session: num of samples per novel (seen) class
              },
```
The number of novel samples is significantly higher than that of seen and old classes. During the first few epochs of training, the accuracy of the New/Unseen category even reaches 99%, before gradually decreasing.

Is this setting intentional? Does it deliberately bias the model toward novel/unseen classes to the maximum extent? Or is there a more balanced way to configure these parameters?

Looking forward to your insights.

**Comment by mashijie1028** [comment]
Hi! This setting is **NOT** intentional, instead, it is a practical setting.

1. We haven't conducted experiments on CIFAR 10 before. We recommend you conduct experiments on CIFAR100, CUB, Tiny and IN100 with our default hyper-parameters.
2. Our basic setup ensures that the number of new and old categories in each continuous learning phase is roughly in the same order of magnitude. However, the specific ratio between seen and unseen classes changes over the course of training, because in our default setting, we keep the number of each class fixed for convenience, and the number of seen classes becomes relatively larger at later continual stages.
3. By default, we sample 50% of all classes as the initial old classes. So on CIFAR10, you might try `offline_old_cls_num=5`. If the initial number of samples in old classes is too large, (i.e., 7 in your example), the number of new class samples becomes larger in continual stages so the evaluation of GCD will naturally tend to favor a higher New ACC. This is beyond our control and is an issue with the evaluation protocol itself.
4. In CUB, IN100 and Tiny, the number of novel samples is relatively smaller, you could try these datasets.
5. In C-GCD, it is practical that the number of samples per novel class is higher than that per old class. Imagine you’re in a university, where most of the knowledge you encounter comes from university-level studies, and only a small portion comes from elementary and middle school education.

**Comment by jinweiisgreat** [comment]
I see. Thank you again! Your work with Professor Liu has been truly inspiring to me.
